### PR TITLE
fix(Android): properly check on pollForValidSSID if network connected

### PR DESCRIPTION
**Pull request by EXio4**
pollForValidSSID is checking if network is 'something', which might just be connecting so it fails to actually wait for a proper connection to be realized - tested with `examples/with-expo` demo on Android 8

### CODE REVIEW

**Comment by EXio4** [comment]
which contributors have time to check this? wondering if there's any issues with the implementation

cc: @JuanSeBestia

**Review by notjulian — COMMENTED** [review]
lgtm

**Comment by DavideViolante** [comment]
LGTM, can you fix the conflicts please?

File: android/src/main/java/com/reactlibrary/rnwifi/RNWifiModule.java [hunk 1]
```diff
@@ -287,92 +287,25 @@ public void connectToProtectedWifiSSID(@NonNull ReadableMap options, final Promi
...
-            }
-
-            if (isWpa3) {
-                builder.setWpa3Passphrase(password);
-            } else if (!password.isEmpty()) {
-                builder.setWpa2Passphrase(password);
-            }
-
-            suggestionsList.add(builder.build());
+    private boolean getConnectionStatus() {
```
Comment (DavideViolante) [inline android/src/main/java/com/reactlibrary/rnwifi/RNWifiModule.java:290]:
```suggestion

    /**
     * Use this to suggest a list of Wi-Fi networks on Android.
     * The promise will resolve with the message 'connected' when the suggestions are added successfully.
     * This method only works for Android and requires a minimum SDK version of 29.
     *
     * @param networkConfigs list of network configurations containing SSID, password, WPA3 flag, and app interaction flag
     * @param promise        to send success/error feedback
     */
    @ReactMethod
    public void suggestWifiNetwork(@NonNull final ReadableArray networkConfigs, final Promise promise) {
        if (!assertLocationPermissionGranted(promise)) {
            return;
        }

        List<WifiNetworkSuggestion> suggestionsList = new ArrayList<>();

        for (int i = 0; i < networkConfigs.size(); i++) {
            ReadableMap config = networkConfigs.getMap(i);
            if (config == null) {
                continue;
            }

            String ssid = config.getString("ssid");
            String password = config.hasKey("password") ? config.getString("password") : "";
            boolean isWpa3 = config.hasKey("isWpa3") && config.getBoolean("isWpa3");
            boolean isAppInteractionRequired = config.hasKey("isAppInteractionRequired") && config.getBoolean("isAppInteractionRequired");

            WifiNetworkSuggestion.Builder builder = new WifiNetworkSuggestion.Builder()
                    .setSsid(ssid);

            if (isAppInteractionRequired) {
                builder.setIsAppInteractionRequired(true); // Conditional (Needs location permission)
            }

            if (isWpa3) {
                builder.setWpa3Passphrase(password);
            } else if (!password.isEmpty()) {
                builder.setWpa2Passphrase(password);
            }

            suggestionsList.add(builder.build());
        }

        final WifiManager wifiManager = (WifiManager) getReactApplicationContext().getSystemService(Context.WIFI_SERVICE);
        final int status = wifiManager.addNetworkSuggestions(suggestionsList);

        if (status != WifiManager.STATUS_NETWORK_SUGGESTIONS_SUCCESS) {
            promise.reject(ConnectErrorCodes.unableToConnect.toString(), "Failed to add network suggestions.");
            return;
        }

        // Optional (Wait for post connection broadcast to one of your suggestions)
        final IntentFilter intentFilter = new IntentFilter(WifiManager.ACTION_WIFI_NETWORK_SUGGESTION_POST_CONNECTION);
        final BroadcastReceiver broadcastReceiver = new BroadcastReceiver() {
            @Override
            public void onReceive(Context context, Intent intent) {
                if (!WifiManager.ACTION_WIFI_NETWORK_SUGGESTION_POST_CONNECTION.equals(intent.getAction())) {
                    promise.reject("failed to connect");
                    return;
                }
                promise.resolve("connected");
            }
        };
        getReactApplicationContext().registerReceiver(broadcastReceiver, intentFilter);
    }

    private boolean getConnectionStatus() {
```

**Comment by DavideViolante** [comment]
Do you know if this fixes #356 ? @EXio4

**Comment by EXio4** [comment]
my bad, couldn’t look at it before, it should be fixing that issue of the polling being done incorrectly - which is how #356 manifested for us - users were “connected” to networks but had failed to do so, we were using a JS-side workaround on our codebase for the meantime, and I didn’t seem to be getting notifications or reminders 😓

if we are having more issues I’ll try to allocate time to help with conflicts, thanks for getting it together and sorry again for disappearing!

**Pull request merged**